src/Storage/Factory/S3Factory.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace EMS\CommonBundle\Storage\Factory;
+
+use EMS\CommonBundle\Storage\Service\S3Storage;
+use EMS\CommonBundle\Storage\Service\StorageInterface;
+use Psr\Log\LoggerInterface;
+
+class S3Factory implements StorageFactoryInterface
+{
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    public function createService(array $parameters): ?StorageInterface
+    {
+        if ('s3' !== $parameters['type'] ?? null) {
+            throw new \RuntimeException('The storage service type doesn\'t match \'s3\'');
+        }
+
+        $credentials = $parameters['credentials'] ?? null;
```

Comment:
You might want to check the option resolver component of symfony to verify required attributes in the parameters. @Davidmattei implemented this also in the FormSubmissions (Handlers) for the "endpoint" input in the handle functions.

Reply:
Done